internal/internal_task_pollers.go

```diff
@@ -1011,24 +1036,32 @@ func convertActivityResultToRespondRequestByID(identity, namespace, workflowID,
 			Identity:   identity}
 	}
 
-	var canceledErr *CanceledError
-	if errors.As(err, &canceledErr) {
-		return &workflowservice.RespondActivityTaskCanceledByIdRequest{
-			Namespace:  namespace,
-			WorkflowId: workflowID,
-			RunId:      runID,
-			ActivityId: activityID,
-			Details:    convertErrDetailsToPayloads(canceledErr.details, dataConverter),
-			Identity:   identity}
+	// Only respond with canceled if allowed
+	if cancelAllowed {
```

Comment:
Is there a way to de-dupe this code?

Reply:
Not really. There isn't as much duplicate code as it might appear (the created structs are different).